aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -884,6 +885,21 @@ Tensor& lcm_(Tensor& self, const Tensor& other) {
   return at::lcm_out(self, self, other);
 }
 
+Tensor& hypot_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  auto iter = TensorIterator::binary_op(result, self, other);
+  hypot_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor hypot(const Tensor& self, const Tensor& other) {
+  Tensor result = at::empty({0}, self.options());
```

Comment:
This has to be constructed as just `Tensor result;` because it's a binary op which, by default, participates in type promotion. If self has dtype torch.float32, for example, and other has dtype torch.float64, then the result type should be torch.float64 when not specified by the user. This would tell TensorIterator that the result type should be torch.float32, however.